### PR TITLE
One node local network

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1175 https://github.com/ava-labs/avalanche-cli/issues/1176

### CODE REVIEW

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -148,6 +156,7 @@ const (
 	ConfigAPMAdminAPIEndpointKey = "admin-api-endpoint"
 	ConfigNodeConfigKey          = "node-config"
 	ConfigMetricsEnabledKey      = "MetricsEnabled"
+	Config1NodeEnabledKey        = "1NodeEnabled"
```
Comment (arturrez) [inline pkg/constants/constants.go:159]: replace 1Node with SingleNode

Comment (arturrez) [reply pkg/constants/constants.go:159]: applicable for everything with `1Node`

**Review by arturrez — APPROVED** [review]

File: pkg/subnet/local.go [hunk 2]
```diff
@@ -676,8 +685,11 @@ func getExpectedDefaultSnapshotSHA256Sum() (string, error) {
 
 // Initialize default snapshot with bootstrap snapshot archive
 // If force flag is set to true, overwrite the default snapshot if it exists
-func SetDefaultSnapshot(snapshotsDir string, force bool) error {
+func SetDefaultSnapshot(snapshotsDir string, force bool, oneNode bool) error {
```
Comment (arturrez) [inline pkg/subnet/local.go:688]: nit: `isSingleNode`

File: cmd/configcmd/singleNode.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/spf13/cobra"
+)
+
+// avalanche config singlenode command
+func newSingleNodeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "singlenode [enable | disable]",
+		Short:        "opt in or out of single node local network",
+		Long:         "set user preference between single node and five nodes local network",
+		RunE:         handleSingleNodeSettings,
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/configcmd/singleNode.go:18]: the use case is faster local dev iteration. a 1 node network is much faster to start and stop. requirements where mentioned by @JacobEverly

[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/configcmd/singleNode.go:18]: as we base fast deploy on pre-made snapshots, and network snapshots are related to a specific number of nodes, it is not possible to provide x number of nodes without have one snapshot for each option. we can do provide x number of nodes by loosing fast deploy for the ones != [1,5] but that involves pretty much more code. I believe it is just good for the moment to provide the 1 node option for the users that demanded that, and we can add x number of nodes as option (and other local network options) for a future milestone

**Pull request merged**